src/v/utils/fragmented_vector.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "vassert.h"
+
+#include <cstddef>
+#include <vector>
+
+/**
+ * A very very simple fragmented vector that provides random access like a
+ * vector, but does not store its data in contiguous memory.
+ *
+ * There is no reserve method because we allocate a full fragment at a time.
+ * However, after you populate a vector you might want to call shrink to fit if
+ * your fragment is large. A more advanced strategy could allocate capacity up
+ * front which just requires a bit more accounting.
+ *
+ * The iterator implementation works for a few things like std::lower_bound,
+ * upper_bound, distance, etc... see fragmented_vector_test.
+ */
+template<typename T, size_t fragment_size = 8192>
```

Comment:
can you comment on the magic number. 

should you have an alias for `small_fragvector`

Reply:
mm, small enough to be in the small allocation pool and provide granularity allocations. still big enough to feed the cache line. I don't have specific performance numbers